elastic/beats
# [Metricbeat] Add FailOnRequired when applying schema

Original problem reported is when running metricbeat mongodb module, `metrics` metricset failled with:
```
Error fetching data for metricset mongodb.metrics: failed to apply schema: 5 errors: key `metrics.repl.apply.attemptsToBecomeSecondary` not found; key `metrics.repl.initialSync` not found; key `metrics.repl.executor` not found; key `metrics.commands` not found; key `metrics.operation.writeConflicts` not found
```
I was able to reproduce it and turned out this error message changes with different keys not found when I reproduced it. 
This error should be fixed by adding `schema.FailOnRequired` when applying the schema.

Second problem is after adding `schema.FailOnRequired`, a warning showed up when trying to run `metrics` metricset:
```
...
"reason":"failed to parse field [mongodb.metrics.query_executor.scanned_indexes] of type [long] in document...
```
I believe some metrics in `metrics` metricset are missing `.count` suffix,  like `mongodb.metrics.query_executor.scanned_indexes`.

## How to test this:
Please start mongodb locally and start Metricbeat with mongodb module enabled. Make sure metrics are flowing with no error above. You might see some error/warning message at INFO level indicating permission error but that's not related to this.
